### ISSUE TITLE
Error when publishing new version of project: 'NoneType' object has no attribute 'email'

### ISSUE THREAD

**Issue by tompollard** [MIT-LCP/physionet-build#1419]
The following error was raised when attempting to publish the new version of a project by clicking "Publish submission" at https://physionet.org/console/submitted-projects/SLUG/publish/:

```
Internal Server Error: /console/submitted-projects/SLUG/publish/

AttributeError at /console/submitted-projects/SLUG/publish/
'NoneType' object has no attribute 'email'

Traceback:

File "/physionet/python-env/physionet/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
 34.             response = get_response(request)

File "/physionet/python-env/physionet/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
 115.                 response = self.process_exception_by_middleware(e, request)

File "/physionet/python-env/physionet/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
 113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/physionet/python-env/physionet/lib/python3.7/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
 21.                 return view_func(request, *args, **kwargs)

File "./console/views.py" in handling_view
 90.                 return base_view(request, *args, **kwargs)

File "./console/views.py" in publish_submission
 535.                 make_zip=int(publish_form.cleaned_data['make_zip']))

File "./project/modelcomponents/activeproject.py" in publish
 533.                         published_author.corresponding_email = author.corresponding_email.email

Exception Type: AttributeError at /console/submitted-projects/SLUG/publish/
Exception Value: 'NoneType' object has no attribute 'email'
```

**Comment by bemoody** [comment]
At a guess, somebody deleted their associated email address while that
address was still marked as corresponding for an active project?

At a basic level, if the corresponding author has a null corresponding
email for any reason, then check_integrity() should fail and it
shouldn't be possible to submit or publish.

But the weird relationship between user-associated-email and
author-corresponding-email and project-corresponding-email is all very
messy and confusing.

Possibly the (corresponding) author should be able to change their
corresponding address up until the time the project is published, or at
least up until the time they approve the project for publication.

If the project is in a state where the author isn't allowed to change
their address, then they also shouldn't be allowed to *delete* that
address from their account.

(Possibly authors shouldn't be allowed to delete their former
corresponding address even after publication.)

**Comment by tompollard** [comment]
> At a guess, somebody deleted their associated email address while that
address was still marked as corresponding for an active project?

Yeah, I think this is what happened.

> At a basic level, if the corresponding author has a null corresponding
email for any reason, then check_integrity() should fail and it
shouldn't be possible to submit or publish.

I agree, this issue should be flagged in the integrity checks at a minimum.

> But the weird relationship between user-associated-email and
author-corresponding-email and project-corresponding-email is all very
messy and confusing.

Confusing, I agree!

For this particular project, I fixed the problem with the following steps:

```
from project.models import ActiveProject

slug="SLUG"
project = ActiveProject.objects.get(slug=slug)
author = project.authors.get(is_corresponding=True)
email = author.user.associated_emails.all()[0]

author.corresponding_email = email
author.save()
```

**Comment by bemoody** [comment]
I haven't directly checked this but it should be fixed by 5a986dad04df3b04ca122640d8b8abecf62a8616 (pull #1735).
